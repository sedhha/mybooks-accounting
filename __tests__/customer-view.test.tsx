import React from 'react';
import { render } from '@testing-library/react';
import CustomerView from '@/components/Customer/View';
import * as NextRouter from 'next/router';

describe('Customer View Screen', () => {
	beforeEach(() => {
		// Create a Jest mock for the `useRouter` hook
		const mockRouter: NextRouter.NextRouter = {
			route: '/',
			pathname: '/',
			asPath: '/',
			basePath: '',
			isLocaleDomain: false,
			push: jest.fn(),
			replace: jest.fn(),
			reload: jest.fn(),
			back: jest.fn(),
			forward: jest.fn(),
			prefetch: jest.fn(),
			beforePopState: jest.fn(),
			events: {
				on: jest.fn(),
				off: jest.fn(),
				emit: jest.fn(),
			},
			isFallback: false,
			isReady: true,
			isPreview: false,
			query: {},
		};

		// Mock the `useRouter` hook
		jest.spyOn(NextRouter, 'useRouter').mockReturnValue(mockRouter);
	});
	afterAll(() => {
		jest.resetAllMocks();
		jest.clearAllMocks();
	});
	test('renders LoginScreen component', () => {
		const { container } = render(<CustomerView />);
		expect(container).toMatchSnapshot();
	});
});
