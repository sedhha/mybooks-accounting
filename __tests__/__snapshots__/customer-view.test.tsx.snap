// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer View Screen renders LoginScreen component 1`] = `
<div>
  <div
    class="Customer"
  >
    <ul
      class="navList"
    >
      <li
        is-active="true"
      >
        View Request
      </li>
      <li
        is-active="false"
      >
        Raise Request
      </li>
    </ul>
    <div
      class="raise"
    >
      <h1
        class="h1"
      >
        Customer ID: 
      </h1>
      <p
        class="logout"
      >
        Not 
        ?
         
        <span>
          Logout
        </span>
      </p>
      <div
        class="userRequests"
      />
    </div>
  </div>
</div>
`;
